appnav/src/main/kotlin/io/element/android/appnav/room/RoomFlowNode.kt

```diff
@@ -172,10 +175,14 @@ class RoomFlowNode @AssistedInject constructor(
         }
     }
 
-    private fun loadingNode(buildContext: BuildContext) = node(buildContext) {
-        Box(modifier = it.fillMaxSize(), contentAlignment = Alignment.Center) {
-            CircularProgressIndicator()
-        }
+    private fun loadingNode(buildContext: BuildContext) = node(buildContext) { modifier ->
+        val networkStatus by networkMonitor.connectivity.collectAsState()
+        LoadingRoomNodeView(
+            state = LoadingRoomState.Loading,
+            hasNetworkConnection = networkStatus == NetworkStatus.Online,
```

Comment:
This is a bit weird to show the hasNetwork connection here and not in room preview :/